quantum/rgblight.c

```diff
@@ -57,79 +57,117 @@ rgblight_config_t rgblight_config;
 LED_TYPE led[RGBLED_NUM];
 bool rgblight_timer_enabled = false;
 
-void sethsv(uint16_t hue, uint8_t sat, uint8_t val, LED_TYPE *led1) {
-  uint8_t r = 0, g = 0, b = 0, base, color;
+#ifdef RGBW
+void convert_rgb_to_rgbw(uint8_t* r, uint8_t* g, uint8_t* b, uint8_t* w) {
+    // Determine lowest value in all three colors, put that into
+    // the white channel and then shift all colors by that amount
+    *w = MIN(*r, MIN(*g, *b));
+    *r -= *w;
+    *g -= *w;
+    *b -= *w;
+}
+#endif
 
-  if (val > RGBLIGHT_LIMIT_VAL) {
-      val=RGBLIGHT_LIMIT_VAL; // limit the val
-  }
+void convert_hsv_to_rgb(uint16_t hue, uint8_t sat, uint8_t val, uint8_t *r_p, uint8_t *g_p, uint8_t *b_p) {
+    uint8_t r, g, b, base, color;
 
-  if (sat == 0) { // Acromatic color (gray). Hue doesn't mind.
-    r = val;
-    g = val;
-    b = val;
-  } else {
-    base = ((255 - sat) * val) >> 8;
-    color = (val - base) * (hue % 60) / 60;
-
-    switch (hue / 60) {
-      case 0:
-        r = val;
-        g = base + color;
-        b = base;
-        break;
-      case 1:
-        r = val - color;
-        g = val;
-        b = base;
-        break;
-      case 2:
-        r = base;
-        g = val;
-        b = base + color;
-        break;
-      case 3:
-        r = base;
-        g = val - color;
-        b = val;
-        break;
-      case 4:
-        r = base + color;
-        g = base;
-        b = val;
-        break;
-      case 5:
-        r = val;
-        g = base;
-        b = val - color;
-        break;
+    if (val > RGBLIGHT_LIMIT_VAL) {
+        val=RGBLIGHT_LIMIT_VAL; // limit the val
     }
-  }
-  r = pgm_read_byte(&CIE1931_CURVE[r]);
-  g = pgm_read_byte(&CIE1931_CURVE[g]);
-  b = pgm_read_byte(&CIE1931_CURVE[b]);
 
+    if (sat == 0) { // Acromatic color (gray). Hue doesn't mind.
+      r = val;
+      g = val;
+      b = val;
+    } else {
+      while(hue > 360) { hue -= 360; } //Making sure Hue is at or below 360
```

Comment:
I believe this is just
```suggestion
      hue %= 360;
```

Reply:
You're right. That's a brain fart on my end.